consensus/ibft/state.go

```diff
@@ -282,8 +286,17 @@ func (v *ValidatorSet) Includes(addr types.Address) bool {
 	return v.Index(addr) != -1
 }
 
-// MaxFaultyNodes returns the maximum number of allowed faulty nodes, based on the current validator set
+// MaxFaultyNodes returns the maximum number of allowed faulty nodes (F), based on the current validator set
 func (v *ValidatorSet) MaxFaultyNodes() int {
-	// numberOfValidators / 3
-	return int(math.Ceil(float64(len(*v))/3)) - 1
+	// N -> number of nodes in IBFT
+	// F -> number of faulty nodes
+	//
+	// N = 3F + 1
+	// => F = (N - 1) / 3
+	//
+	// IBFT tolerates 1 failure with 4 nodes
+	// 4 = 3 * 1 + 1
+	// To tolerate 2 failures, IBFT requires 7 nodes
+	// 7 = 3 * 2 + 1
+	return int(math.Floor(float64(len(*v)-1) / 3))
```

Comment:
This can be reduced to `(len(arr)-1) / 3`.

Reply:
Yes, it's simpler, but for a length of 0 it doesn't bind it to 0 (it goes negative):
https://goplay.tools/snippet/R3en9z64S7F

Reply:
I just noticed I've swapped the outputs 🤦 

Fixed 👍 

2df2d26

Reply:
GG :)